### PR TITLE
Fix traffic processing

### PR DESCRIPTION
**Pull request by syershov**

### CODE REVIEW

File: traffic/traffic_info.cpp [hunk 1]
```diff
@@ -138,10 +138,17 @@ void TrafficInfo::SetTrafficKeysForTesting(vector<RoadSegmentId> const & keys)
 bool TrafficInfo::ReceiveTrafficData(string & etag)
 {
   vector<SpeedGroup> values;
-  if (!ReceiveTrafficValues(etag, values))
-    return false;
-
-  return UpdateTrafficData(values);
+  switch(ReceiveTrafficValues(etag, values))
```
Comment (mpimenov) [inline traffic/traffic_info.cpp]: clang-format

File: traffic/traffic_info.cpp [hunk 2]
```diff
@@ -398,17 +405,17 @@ bool TrafficInfo::ReceiveTrafficKeys()
   return true;
 }
 
-bool TrafficInfo::ReceiveTrafficValues(string & etag, vector<SpeedGroup> & values)
+TrafficInfo::DataState TrafficInfo::ReceiveTrafficValues(string & etag, vector<SpeedGroup> & values)
```
Comment (mpimenov) [inline traffic/traffic_info.cpp]: s/TrafficInfo:://

File: traffic/traffic_info.cpp (hunk 1 above)
Comment (syershov) [inline traffic/traffic_info.cpp]: done

File: traffic/traffic_info.hpp [hunk 3]
```diff
@@ -127,12 +135,12 @@ class TrafficInfo
   // Returns true and updates m_coloring if the values are read successfully and
   // their number is equal to the number of keys.
   // Otherwise, returns false and does not change m_coloring.
-  bool ReceiveTrafficValues(string & etag, vector<SpeedGroup> & values);
+  DataState ReceiveTrafficValues(string & etag, vector<SpeedGroup> & values);
```
Comment (mpimenov) [inline traffic/traffic_info.hpp]: The documentation to this method is wrong.

File: traffic/traffic_info.cpp (hunk 2 above)
Comment (mpimenov) [inline traffic/traffic_info.cpp]: Can't you update m_dataState and check it?
Overall, what we do now is VERY dirty: we basically have 3 separate classes for
data availability, two here and one in traffic manager.

File: traffic/traffic_info.hpp (hunk 3 above)
Comment (syershov) [inline traffic/traffic_info.hpp]: dщту

Comment (syershov) [inline traffic/traffic_info.hpp]: done

File: traffic/traffic_info.cpp (hunk 2 above)
Comment (syershov) [inline traffic/traffic_info.cpp]: done

**Comment by syershov** [comment]
@mpimenov PTAL

File: traffic/traffic_info.hpp [hunk 4]
```diff
@@ -117,22 +117,29 @@ class TrafficInfo
   static void DeserializeTrafficValues(vector<uint8_t> const & data, vector<SpeedGroup> & result);
 
 private:
+  enum class ServerData
```
Comment (mpimenov) [inline traffic/traffic_info.hpp]: ServerDataStatus?

Comment (syershov) [inline traffic/traffic_info.hpp]: Done

**Comment by syershov** [comment]
@mpimenov PTAL

File: traffic/traffic_info.hpp [hunk 5]
```diff
@@ -117,22 +117,29 @@ class TrafficInfo
...
 
   // todo(@m) A temporary method. Remove it once the keys are added
   // to the generator and the data is regenerated.
   bool ReceiveTrafficKeys();
 
-  // Tries to read the values of the Coloring map from server.
-  // Returns true and updates m_coloring if the values are read successfully and
-  // their number is equal to the number of keys.
+  // Tries to read the values of the Coloring map from server into |values|.
+  // Returns result of communicating with server as DataState.
```
Comment (syershov) [inline traffic/traffic_info.hpp]: done

**Comment by mpimenov** [comment]
LGTM

**Pull request merged**